### PR TITLE
Teams zod validation and IT

### PR DESCRIPTION
**Pull request by darshangodase**
## Description

Fixed zod validation of teams routes  and added integration tests for team routes covering all positive ,negative tests
- Teams zod vadations
- Teams IT
- Add azure openai,gemini,grok model provider support in Integration tests
- Added interface for api client in response validation for all modules
- Fixed upload record default isversioned from false to true 

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

### CODE REVIEW

File: backend/nodejs/apps/src/modules/user_management/validators/teams.validators.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      validateNoFormatSpecifiers(value, 'search parameter');
+    } catch (error: unknown) {
+      const message =
+        error instanceof Error ? error.message : 'Invalid search parameter';
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message,
+      });
+    }
+  });
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/nodejs/apps/src/modules/user_management/validators/teams.validators.ts]: fixed

File: backend/nodejs/apps/src/modules/user_management/validators/teams.request.validators.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  z.number().int().positive().optional(),
+);
+
+const normalizeOptionalSearch = (arg: unknown): string | undefined => {
+  if (arg === undefined || arg === '') {
+    return undefined;
+  }
+  const trimmed = String(arg).trim();
+  return trimmed === '' ? undefined : trimmed;
+};
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/nodejs/apps/src/modules/user_management/validators/teams.request.validators.ts:61]: fixed

File: backend/nodejs/apps/src/modules/user_management/validators/teams.request.validators.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+const teamLimitSchema = z.preprocess(
+  (arg) => (arg === undefined || arg === '' ? undefined : Number(arg)),
+  z.number().min(1).max(100).default(10),
+);
+
+const timestampQuerySchema = z.preprocess(
+  (arg) => (arg === undefined || arg === '' ? undefined : Number(arg)),
+  z.number().int().positive().optional(),
+);
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/nodejs/apps/src/modules/user_management/validators/teams.request.validators.ts:52]: fixed

File: backend/nodejs/apps/tests/modules/user_management/routes/teams.routes.coverage.test.ts [hunk 4]
```diff
@@ -58,14 +58,6 @@ describe('Teams Routes - handler coverage', () => {
       await handler({ body: { name: 'Team1' } } as any, mockRes(), sinon.stub())
       expect(mockTeamsController.createTeam.calledOnce).to.be.true
     })
-
-    it('should call next on error', async () => {
```
Comment (rish664) [inline backend/nodejs/apps/tests/modules/user_management/routes/teams.routes.coverage.test.ts:62]: why is this removed?

File: backend/nodejs/apps/tests/modules/user_management/routes/teams.routes.test.ts [hunk 5]
```diff
@@ -49,12 +49,6 @@ describe('Teams Routes', () => {
     expect(router).to.be.a('function');
   });
 
-  it('should have route handlers registered', () => {
```
Comment (rish664) [inline backend/nodejs/apps/tests/modules/user_management/routes/teams.routes.test.ts:52]: why is this removed?

File: backend/nodejs/apps/tests/modules/user_management/routes/teams.routes.test.ts [hunk 6]
```diff
@@ -146,47 +139,22 @@ describe('Teams Routes', () => {
     });
   });
 
-  describe('middleware chains', () => {
```
Comment (rish664) [inline backend/nodejs/apps/tests/modules/user_management/routes/teams.routes.test.ts:149]: why is this removed?

File: backend/nodejs/apps/src/modules/user_management/validators/teams.request.validators.ts [hunk 7]
```diff
@@ -0,0 +1,197 @@
+import { z } from 'zod';
```
Comment (rish664) [inline backend/nodejs/apps/src/modules/user_management/validators/teams.request.validators.ts:1]: the name of the file does not looks correct, is it just validating the request params?

File: integration-tests/response-validation/user_management/integration_test_teams.py [hunk 8]
```diff
@@ -0,0 +1,921 @@
...
+
+
+# ---------------------------------------------------------------------------
+# HTTP helpers
+# ---------------------------------------------------------------------------
+def _url(client: PipeshubClient, path: str = "") -> str:
+    return f"{client.base_url}{_BASE_PATH}{path}"
+
+
+def _request(
```
Comment (rish664) [inline integration-tests/response-validation/user_management/integration_test_teams.py]: use the existing way to make http request and if does not exist, define it at a proper place so it can be used by every IT

File: integration-tests/response-validation/user_management/integration_test_teams.py [hunk 9]
```diff
@@ -0,0 +1,921 @@
...
+        method,
+        _url(client, path),
+        headers=headers,
+        json=json,
+        params=params,
+        timeout=client.timeout_seconds,
+    )
+
+
+def _get(client: PipeshubClient, path: str = "", **kw: Any) -> requests.Response:
```
Comment (rish664) [inline integration-tests/response-validation/user_management/integration_test_teams.py]: this is not correct way to define HTTP Verb methods at IT

File: integration-tests/response-validation/user_management/integration_test_teams.py [hunk 10]
```diff
@@ -0,0 +1,921 @@
...
+    return _request(client, "GET", path, **kw)
+
+
+def _post(client: PipeshubClient, path: str = "", **kw: Any) -> requests.Response:
```
Comment (rish664) [inline integration-tests/response-validation/user_management/integration_test_teams.py]: this is not correct way to define HTTP Verb methods at IT

File: integration-tests/response-validation/user_management/integration_test_teams.py [hunk 11]
```diff
@@ -0,0 +1,921 @@
...
+    return _request(client, "POST", path, **kw)
+
+
+def _put(client: PipeshubClient, path: str = "", **kw: Any) -> requests.Response:
```
Comment (rish664) [inline integration-tests/response-validation/user_management/integration_test_teams.py]: same

File: integration-tests/response-validation/user_management/integration_test_teams.py [hunk 12]
```diff
@@ -0,0 +1,921 @@
...
+    return _request(client, "PUT", path, **kw)
+
+
+def _delete_request(client: PipeshubClient, path: str = "", **kw: Any) -> requests.Response:
```
Comment (rish664) [inline integration-tests/response-validation/user_management/integration_test_teams.py]: same

File: backend/nodejs/apps/src/modules/user_management/validators/teams.request.validators.ts [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+});
+
+export const getTeamUsersSchema = z.object({
+  params: teamIdParams,
+  query: teamListQueryObject,
+});
+
+export const getUserTeamsQuerySchema = z.object({
+  query: teamListQueryObject.extend({
+    created_by: mongoUserId.optional(),
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/nodejs/apps/src/modules/user_management/validators/teams.request.validators.ts]: fixed

File: integration-tests/helper/pipeshub_client.py [hunk 14]
```diff
@@ -266,6 +266,51 @@ def _url(self, path: str) -> str:
...
+
+        for attempt in range(2):
+            if auth:
+                self._ensure_access_token()
+                extra_headers = kwargs.pop("headers", None) or {}
+                headers = self._headers()
+                # Let requests set multipart boundary when uploading files.
+                if kwargs.get("files"):
+                    headers.pop("Content-Type", None)
+                kwargs["headers"] = {**headers, **extra_headers}
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply integration-tests/helper/pipeshub_client.py:303]: fixed

File: integration-tests/response-validation/user_management/integration_test_teams.py [hunk 15]
```diff
@@ -0,0 +1,882 @@
...
+                continue
+            uid = str(u.get("userId") or "")
+            if uid and uid not in seen:
+                seen.append(uid)
+                if min_count is not None and len(seen) >= min_count:
+                    return seen[:min_count]
+        if not body.get("pagination", {}).get("hasNext"):
+            break
+        page += 1
+    return seen
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply integration-tests/response-validation/user_management/integration_test_teams.py:92]: fixed

File: integration-tests/response-validation/user_management/integration_test_teams.py (hunk 12 above)
Comment (darshangodase) [reply integration-tests/response-validation/user_management/integration_test_teams.py]: done

File: integration-tests/response-validation/user_management/integration_test_teams.py (hunk 11 above)
Comment (darshangodase) [reply integration-tests/response-validation/user_management/integration_test_teams.py]: done

File: integration-tests/response-validation/user_management/integration_test_teams.py (hunk 10 above)
Comment (darshangodase) [reply integration-tests/response-validation/user_management/integration_test_teams.py]: done

File: integration-tests/response-validation/user_management/integration_test_teams.py (hunk 9 above)
Comment (darshangodase) [reply integration-tests/response-validation/user_management/integration_test_teams.py]: done

File: integration-tests/response-validation/user_management/integration_test_teams.py (hunk 8 above)
Comment (darshangodase) [reply integration-tests/response-validation/user_management/integration_test_teams.py]: done

File: backend/nodejs/apps/src/modules/user_management/validators/teams.request.validators.ts (hunk 7 above)
Comment (darshangodase) [reply backend/nodejs/apps/src/modules/user_management/validators/teams.request.validators.ts:1]: fixed

File: backend/nodejs/apps/tests/modules/user_management/routes/teams.routes.coverage.test.ts (hunk 4 above)
Comment (darshangodase) [reply backend/nodejs/apps/tests/modules/user_management/routes/teams.routes.coverage.test.ts:62]: it is redudant

File: backend/nodejs/apps/tests/modules/user_management/routes/teams.routes.test.ts (hunk 6 above)
Comment (darshangodase) [reply backend/nodejs/apps/tests/modules/user_management/routes/teams.routes.test.ts:149]: added back

File: backend/nodejs/apps/tests/modules/user_management/routes/teams.routes.test.ts (hunk 5 above)
Comment (darshangodase) [reply backend/nodejs/apps/tests/modules/user_management/routes/teams.routes.test.ts:52]: it is redudant,and much stronger test cases with exact count of routes there in code

**Review by vishwjeet-pawar03 — APPROVED** [review]
